assets/src/edit-story/components/canvas/karma/multiSelectionMovable.karma.js

```diff
@@ -0,0 +1,171 @@
+import { TEXT_ELEMENT_DEFAULT_FONT } from '../../../app/font/defaultFonts';
+
+describe('Multi-selection Movable integration', () => {
+  let fixture;
+
+  beforeEach(async () => {
+    fixture = new Fixture();
+
+    await fixture.render();
+  });
+
+  afterEach(() => {
+    fixture.restore();
+  });
+
+  describe('multi-selection', () => {
+    let element1;
+    let element2;
+    let element3;
+    let frame1;
+    let frame2;
+    let frame3;
+
+    beforeEach(async () => {
+      const insertElement = await fixture.renderHook(() => useInsertElement());
+      element1 = await fixture.act(() =>
+        insertElement('text', {
+          font: TEXT_ELEMENT_DEFAULT_FONT,
+          content: 'Text A',
+          x: 10,
+          y: 10,
+          width: 50,
+          height: 50,
+        })
+      );
+
+      element2 = await fixture.act(() =>
+        insertElement('text', {
+          font: TEXT_ELEMENT_DEFAULT_FONT,
+          content: 'Text B',
+          x: 100,
+          y: 100,
+          width: 50,
+          height: 50,
+        })
+      );
+
+      element3 = await fixture.act(() =>
+        insertElement('text', {
+          font: TEXT_ELEMENT_DEFAULT_FONT,
+          content: 'Text B',
+          x: element1.x + element1.width + 1,
+          y: element1.y + element1.height + 1,
+          width: 20,
+          height: 20,
+        })
+      );
+      frame3 = fixture.querySelector(
+        `[data-element-id="${element3.id}"] [data-testid="textFrame"]`
+      );
+
+      frame1 = fixture.querySelector(
+        `[data-element-id="${element1.id}"] [data-testid="textFrame"]`
+      );
+
+      frame2 = fixture.querySelector(
+        `[data-element-id="${element2.id}"] [data-testid="textFrame"]`
+      );
+    });
+
+    it('should render initial content', () => {
+      expect(frame1.textContent).toEqual('Text A');
+      expect(frame2.textContent).toEqual('Text B');
+    });
+
+    it('should multi-select elements on Shift + click', async () => {
+      await fixture.events.click(frame1);
+      await fixture.events.keyboard.down('Shift');
+      await fixture.events.click(frame2);
+      await fixture.events.keyboard.up('Shift');
+
+      const storyContext = await fixture.renderHook(() => useStory());
+      expect(storyContext.state.selectedElementIds).toEqual([
+        element1.id,
+        element2.id,
+      ]);
+    });
+
+    describe('click interaction', () => {
+      let safezone;
+      beforeEach(async () => {
+        await fixture.events.click(frame1);
+        await fixture.events.keyboard.down('Shift');
+        await fixture.events.click(frame2);
+        await fixture.events.keyboard.up('Shift');
+
+        safezone = fixture.querySelector('[data-testid="safezone"]');
+      });
+
+      it('should select one element when clicking in multi-selection', async () => {
+        const { x, y } = safezone.getBoundingClientRect();
+        await fixture.events.mouse.click(
+          x + element1.x + 1,
```

Comment:
This goes straight to the zoomed-vs-unzoomed editor nuances. Any thoughts how we could be addressing this? We could:
1. Always start puppeteer with a big enough viewport to fit the unzoomed editor, but then we will be missing zoomed tests.
2. Run all tests automatically with two zoom factors: unzoomed and zoomed. But it might be hard to write tests?

Reply:
For me, it always was a bigger chance that something will work on a big canvas, so I would say test on a small canvas (if only, cause I found most of the bugs there).